### DISCUSSION TITLE
Would dagger be open to replacing net/http with something like hashicorp/go-retryablehttp?

### DISCUSSION THREAD

**Discussion by mdgreenwald** [dagger/dagger#9256] [Ideas]
## What ##
Replacing `net/http` with `hashicorp/go-retryablehttp` or something like it.

> "The `retryablehttp` package provides a familiar HTTP client interface with **automatic retries and exponential backoff**"

## Why ##
Seems like a really good thing to have given how flaky CI/CD can often be by nature of where, what, and how it runs. e.g. Pushing artifacts to remote servers, sometimes over the public internet. Sometimes there's a thundering herd that causes a failure. Etc... Built in retries would make this less of a problem for everyone and would be more efficient than ad-hoc retry logic.

We use dagger and see many transient errors on our pipelines and it would be nice to have built in retries in dagger instead of having to add them ad-hoc throughout our pipelines whenever we see them cropping up.

## Who ##
Me, I can work on opening a PR to dagger to do this. Or anyone else could too.

**Comment by marcosnils** [comment]
> We use dagger and see many transient errors on our pipelines and it would be nice to have built in retries in dagger instead of having to add them ad-hoc throughout our pipelines whenever we see them cropping up.

Interesting observation. In our personal use of Dagger which is quite extensive, I haven't seen net/http errors that often. Is there any particular operation through the net/http package that you generally see failing often in your particular case?

**Reply by jedevc** [reply]
Yeah, potentially open to it :tada: The problem is that lots of the places this would be valuable would be in our use of buildkit.

So, yeah kinda curious where you see this, to see where we'd specifically want to change.
